modules/references/server/controllers/reference.server.controller.js

```diff
@@ -404,14 +404,24 @@ exports.readMany = async function readMany(req, res, next) {
     const selfId = req.user._id;
 
     const userToId = new mongoose.Types.ObjectId(userTo);
-    const matchQuery = {
+    let matchQuery = {
       $or: [{ userTo: userToId }, { userFrom: userToId }],
     };
     // Allow non-public references only when userTo is self
     if (!selfId.equals(userToId)) {
       matchQuery.public = true;
     }
 
+    const privateFromSelfQuery = {
+      userFrom: selfId,
+      userTo: userToId,
+      public: false,
+    };
+
+    matchQuery = {
+      $or: [matchQuery, privateFromSelfQuery],
+    };
```

Comment:
I find the `matchquery` reshuffling a bit hard to grok.

How about something more simple, like so:
```js
    const matchQuery = {
      $or: [
        // Experiences to/from this member
        {
          $or: [{ userTo: userToId }, { userFrom: userToId }],
        },
        // Authenticated user's pending references to this member
        {
          userFrom: selfId,
          userTo: userToId,
          public: false,
        },
      ],
    };

    // Allow non-public references only when userTo is self
    if (!selfId.equals(userToId)) {
      matchQuery.public = true;
    }
```

What do you think?

Reply:
I agree. Should we, though, put all the queries inside one `$or`?

Reply:
Yeah, I think that would work. 👍

Reply:
Not sure where exactly that `matchQuery.public = true;` would need to be though, inside first block? 🤔

Reply:
How about doing it like this:

```
function buildReadManyQuery(userToId, selfId) {
  const allExperiencesFromOrToUser = {
    $or: [{ userTo: userToId }, { userFrom: userToId }],
  };

  const privacyRestrictions = {
    $or: [
      // all published experiences:
      { public: true },
      // pending experiences only from or to Authenticated user:
      { public: false, userTo: selfId },
      { public: false, userFrom: selfId },
    ],
  };

  return {
    $and: [allExperiencesFromOrToUser, privacyRestrictions],
  };
}

```